lib/BasicResolver.php

```diff
@@ -77,8 +80,16 @@ public function __destruct() {
         Loop::cancel($this->gcWatcher);
     }
 
+    /**
+     * @param int $typeRestriction
+     */
+    public function setTypeRestriction(int $typeRestriction) {
+        $this->typeRestriction = $typeRestriction;
+    }
```

Comment:
Should this be a setter or injected as constructor parameter?

Reply:
Actually I don't care. Tell me what better fits your project design. When using a setter you can change it without needing to create a new resolver.

Reply:
Yes, it's just not something we currently do very often. I'd like to hear the opinion of @bwoebi and @trowski.

Reply:
I would prefer to not use a setter for this. You don't want to make it mutable throughout your code base at random places.

If you need to change the restricted types just create a new instance instead of changing "global" state.

Reply:
I agree, a ctor argument fits better here. Perhaps a configuration object for further extension?

Reply:
We already have a config object. https://github.com/amphp/dns/blob/master/lib/Config.php

Reply:
... which is generated by the ConfigLoader and not directly configurable by the user.

Reply:
Is there a way we can detect if a system has a IPv6 address?

Reply:
@trowski See https://github.com/amphp/socket/issues/35.

Reply:
Ah, too bad. Well then +1 to a ctor argument.

Reply:
Did you recognize my latest changes?

Reply:
Yes. Did you see [my comment](https://github.com/amphp/dns/pull/67#issuecomment-342509701)? Doesn't that solve the problem already?

Reply:
Yes i saw them. Hm kind of. It does actually help but the error message is confusing. If i get a connect error on a ipv4 entry, it skips over to a v6 record and this fails due to "network unreachable" error but the original v4 error is masked by the v6 connect error. So it does not really break (v4 connect failed anyway) but the v6 error is not really transparent

Reply:
We could improve the error message to mention previously failed attempts. I prefer that if that fully solves the issue over adding configuration options.

Reply:
Sounds good! I only realized the v4/6 error message thing after I disabled the v6 records

Reply:
See https://github.com/amphp/dns/pull/67.